packages/task/src/browser/task-service.ts

```diff
@@ -210,27 +211,29 @@ export class TaskService implements TaskConfigurationClient {
             this.runningTasks.get(event.taskId)!.terminateSignal.resolve(event.signal);
             setTimeout(() => this.runningTasks.delete(event.taskId), 60 * 1000);
 
-            const taskConfiguration = event.config;
-            let taskIdentifier = event.taskId.toString();
-            if (taskConfiguration) {
-                taskIdentifier = this.taskNameResolver.resolve(taskConfiguration);
-            }
-
+            const taskConfig = event.config;
+            const taskIdentifier = taskConfig ? this.getTaskIdentifier(taskConfig) : event.taskId.toString();
             if (event.code !== undefined) {
-                const message = `Task ${taskIdentifier} has exited with code ${event.code}.`;
+                const message = `Task '${taskIdentifier}' has exited with code ${event.code}.`;
                 if (event.code === 0) {
                     this.messageService.info(message);
                 } else {
                     this.messageService.error(message);
                 }
             } else if (event.signal !== undefined) {
-                this.messageService.info(`Task ${taskIdentifier} was terminated by signal ${event.signal}.`);
+                this.messageService.info(`Task '${taskIdentifier}' was terminated by signal ${event.signal}.`);
             } else {
                 console.error('Invalid TaskExitedEvent received, neither code nor signal is set.');
             }
         });
     }
 
+    private getTaskIdentifier(taskConfig: TaskConfiguration): string {
+        const taskName = this.taskNameResolver.resolve(taskConfig);
+        const sourceStrUri = this.taskSourceResolver.resolve(taskConfig);
+        return `${taskName} (${new URI(sourceStrUri).displayName})`;
```

Comment:
[minor] do we need to use `labelProvider.getName(uri)` for such a case?:
https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#use-short-name